### PR TITLE
 Add simple POST-command and changed upload path (remote)

### PR DESCRIPTION
**Pull request by deadshot8086**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Closes #1897

### CODE REVIEW

File: test/scripts/http_post_cmd_upload.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
```
Comment (XVilka) [inline test/scripts/http_post_cmd_upload.py:1]: Please add SPDX license information

**Review by XVilka — CHANGES_REQUESTED** [review]
It fails on Windows (AppVeyor):
```
[XX] C:\projects\rizin\test\db\cmd\cmd_http_post http-post-cmd
ANSICON=1 RZ_NOPLUGINS=1 C:\projects\rizin\rizin-vs2022_64-v0.7.0\bin\rizin.exe -escr.utf8=0 -escr.color=0 -escr.interactive=0 -eflirt.sigdb.load.system=false -eflirt.sigdb.load.home=false -N -Qc '!python3 scripts/http_post_cmd_upload.py
' =
-- stdout
--- expected
+++ actual
@@ -1,1 +1,0 @@
-Test succeeded
-- exit status: -1
[**]        C:\projects\rizin\test\db\cmd\cmd_http_post    18648 OK      1020 BR        1 XX       31 FX
```

**Review by XVilka — CHANGES_REQUESTED** [review]
```
clang -Ilibrz/core/librz_core.so.0.7.0.p -I. -I.. -Ilibrz -I../librz -Ilibrz/include -I../librz/include -Ilibrz/core/cmd_descs -I../librz/core/cmd_descs -Ilibrz/util/sdb/src -I../librz/util/sdb/src -I../librz/asm/arch/include -I../librz/asm/arch -I../librz/asm/arch/h8300 -I../librz/asm/arch/hexagon -I../librz/asm/arch/msp430 -I../librz/asm/arch/rsp -I../librz/asm/arch/mcore -I../librz/asm/arch/v850 -I../librz/asm/arch/propeller -I../librz/asm/arch/ebc -I../librz/asm/arch/cr16 -I../librz/asm/arch/8051 -I../librz/asm/arch/v810 -I../librz/asm/arch/or1k -I../librz/asm/arch/tricore -I../librz/analysis/arch/gb -I../librz/type/parser -I../librz/bin/format/elf -I../librz/bin/format/dmp -I../librz/bin/format/mdmp -I../librz/bin/format/pe -I../librz/bin/format -I../subprojects/rzgdb/include -I../subprojects/rzgdb/include/gdbclient -I../subprojects/rzgdb/include/gdbserver -Isubprojects/rzheap -I../subprojects/rzheap -I../subprojects/rzheap/rz_jemalloc -I../subprojects/rzheap/rz_windows -I../subprojects/rizin-shell-parser/src/tree_sitter -I../subprojects/tree-sitter-0.20.7/lib/src -I../subprojects/tree-sitter-0.20.7/lib/include -fdiagnostics-color=always -fsanitize=address,undefined -fno-omit-frame-pointer -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -O2 -g -DRZ_PLUGIN_INCORE=1 -D_GNU_SOURCE --std=gnu99 -Werror=sizeof-pointer-memaccess -fvisibility=hidden -shared-libasan -DASAN=1 -DRZ_ASSERT_STDOUT=1 -Wno-cpp -fPIC -MD -MQ librz/core/librz_core.so.0.7.0.p/rtr.c.o -MF librz/core/librz_core.so.0.7.0.p/rtr.c.o.d -o librz/core/librz_core.so.0.7.0.p/rtr.c.o -c ../librz/core/rtr.c
In file included from ../librz/core/rtr.c:143:
../librz/core/rtr_http.c:130:41: error: variable 'out' is uninitialized when used here [-Werror,-Wuninitialized]
                                rz_core_rtr_http_cmd(core, rs, cmd, out, headers);
                                                                    ^~~
../librz/core/rtr_http.c:126:14: note: initialize the variable 'out' to silence this warning
                                char *out, *cmd = rs->path + 5;
                                         ^
                                          = NULL
1 error generated.
```

**Review by XVilka — CHANGES_REQUESTED** [review]
```
Run export PATH=${HOME}/Library/Python/3.9/bin:${HOME}/Library/Python/3.10/bin:${HOME}/.local/bin:${PATH}
pylint: Command line or configuration file:1: UserWarning: 'BaseException' is not a proper value for the 'overgeneral-exceptions' option. Use fully qualified name (maybe 'builtins.BaseException' ?) instead. This will cease to be checked at runtime when the configuration upgrader is released.
pylint: Command line or configuration file:1: UserWarning: 'Exception' is not a proper value for the 'overgeneral-exceptions' option. Use fully qualified name (maybe 'builtins.Exception' ?) instead. This will cease to be checked at runtime when the configuration upgrader is released.
************* Module fuzz_rz_asm
test/fuzz/scripts/fuzz_rz_asm.py:137:11: E0601: Using variable 'sys' before assignment (used-before-assignment)

-----------------------------------
Your code has been rated at 9.94/10

Error: Process completed with exit code 123.
```

File: librz/core/rtr_http.c [hunk 2]
```diff
@@ -2,6 +2,301 @@
...
+			return rz_core_rtr_http_handler_get_file;
+		} else if (!strncmp(rs->path, "/cmd/", 5)) {
+			return rz_core_rtr_http_handler_get_cmd;
+		} else {
+			return rz_core_rtr_http_handler_get_index;
+		}
+	} else if (!strcmp(rs->method, "POST")) {
+		if (!strncmp(rs->path, "/upload/", 8)) {
+			return rz_core_rtr_http_handler_post_upload;
+		} else if (!strncmp(rs->path, "/cmd/", 5)) {
```
Comment (wargio) [inline librz/core/rtr_http.c]: use `strlen` and let the compiler optimize this.

File: librz/core/rtr_http.c [hunk 3]
```diff
@@ -43,7 +338,7 @@ static int rz_core_rtr_http_run(RzCore *core, int launch, int browse, const char
 	char *dir;
 	int iport;
 	const char *bind = rz_config_get(core->config, "http.bind");
-	const char *index = rz_config_get(core->config, "http.index");
+	// const char *index = rz_config_get(core->config, "http.index");
```
Comment (wargio) [inline librz/core/rtr_http.c]: if not used, remove this line

**Review by wargio — CHANGES_REQUESTED** [review]
LGTM just some comments that should be fixed. also if `http.index` is not used anywhere else, you can also do a cleanup and remove it from `cconfig.c`

File: librz/core/rtr_http.c (hunk 2 above)
Comment (wargio) [reply librz/core/rtr_http.c]: example `(!strncmp(rs->path, "/cmd/", strlen("/cmd/")))`

**Comment by deadshot8086** [comment]
> LGTM just some comments that should be fixed. also if `http.index` is not used anywhere else, you can also do a cleanup and remove it from `cconfig.c`

ya it is used in remote get command in browser to display index.html when no cmd is specified, so can't remove it

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Pull request merged**